+nosnoc/+solver/RelaxationSolver.m

```diff
@@ -54,34 +60,36 @@
                 % TODO(@anton) implement this
             else % Otherwise use vdx internally anyway but be sad about interleaving
                 use_vdx = false;
+                % preprocess empty fields
+                
                 nlp = vdx.Problem('casadi_type', casadi_symbolic_mode);
-                nlp.w.mpcc_w(0) = {mpcc.x};
-                nlp.p.mpcc_p(0) = {mpcc.p};
-                nlp.g.mpcc_g(0) = {mpcc.g};
-                nlp.p.sigma_p(0) = {{'sigma_p', 1}, 0, inf, opts.sigma_0};
+                nlp.w.mpcc_w = {mpcc.x};
+                nlp.p.mpcc_p = {mpcc.p};
+                nlp.g.mpcc_g = {mpcc.g};
+                nlp.p.sigma_p = {{'sigma_p', 1}, opts.sigma_0};
                 nlp.f = mpcc.f;
                 n_c = size(mpcc.G);
 
                 % Create relaxation slacks/parameters
                 switch opts.elasticity_mode
                   case ElasticityMode.NONE
-                    sigma = nlp.p.sigma_p(0);
+                    sigma = nlp.p.sigma_p();
                   case ElasticityMode.ELL_INF
-                    nlp.w.s_elastic(0) = {{'s_elastic', 1}, opts.s_elastic_min, opts.s_elastic_max, opts.s_elastic_0};
-                    sigma = nlp.w.s_elastic(0);
+                    nlp.w.s_elastic = {{'s_elastic', 1}, opts.s_elastic_min, opts.s_elastic_max, opts.s_elastic_0};
```

Comment:
add comment in code: adding elastic mode variable to the existing mpcc variables

Reply:
Added Comment